src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -286,20 +286,28 @@ void GcsResourceManager::Initialize(const GcsInitData &gcs_init_data) {
 }
 
 void GcsResourceManager::OnNodeAdd(const rpc::GcsNodeInfo &node) {
+  NodeID node_id = NodeID::FromBinary(node.node_id());
+  scheduling::NodeID scheduling_node_id(node_id.Binary());
   if (!node.resources_total().empty()) {
-    scheduling::NodeID node_id(node.node_id());
     for (const auto &entry : node.resources_total()) {
       cluster_resource_manager_.UpdateResourceCapacity(
-          node_id, scheduling::ResourceID(entry.first), entry.second);
+          scheduling_node_id, scheduling::ResourceID(entry.first), entry.second);
     }
   } else {
-    RAY_LOG(WARNING) << "The registered node " << NodeID::FromBinary(node.node_id())
+    RAY_LOG(WARNING) << "The registered node " << node_id
                      << " doesn't set the total resources.";
   }
+
+  if (!node.labels().empty()) {
```

Comment:
if labels are empty, shouldn't we just update the node labels to empty?

Reply:
This is the semantics of 'update', so it won't be overwritten. If such a scenario arises in the future, I will create a new interface called 'ResetNodeLabels()'